flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableResult.java

```diff
@@ -49,9 +49,33 @@
 	ResultKind getResultKind();
 
 	/**
-	 * Get the result contents as a row iterator.
+	 * Get the result contents as a closeable row iterator.
+	 *
+	 * <p><strong>NOTE:</strong>If this result corresponds to a flink job,
+	 * the job will not be finished unless all result data has been collected.
+	 * So we should actively close the job to avoid resource leak.
+	 *
+	 * <p>There are two approaches to close a job:
+	 * 1. close the job through JobClient, for example:
```

Comment:
This is actually not recommended. If user cancels the job but the iterator does not know it, the next call to the iterator might throw exception. We should only recommend the users to close the iterator if they don't need more data.